scala/lambdas/ingest-flow-control/src/test/scala/uk/gov/nationalarchives/ingestflowcontrol/LambdaTest.scala

```diff
@@ -0,0 +1,425 @@
+package uk.gov.nationalarchives.ingestflowcontrol
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers.*
+import cats.effect.*
+import cats.effect.unsafe.implicits.global
+import io.circe.{Decoder, Encoder}
+import org.scalatest.EitherValues
+import org.scanamo.DynamoFormat
+import org.scanamo.request.RequestCondition
+import software.amazon.awssdk.services.dynamodb.model.BatchWriteItemResponse
+import software.amazon.awssdk.services.sfn.model.StartExecutionResponse
+import uk.gov.nationalarchives.{DADynamoDBClient, DASFNClient, DASSMClient}
+import uk.gov.nationalarchives.ingestflowcontrol.Lambda.*
+import uk.gov.nationalarchives.dynamoformatters.DynamoFormatters.*
+import uk.gov.nationalarchives.ingestflowcontrol.helpers.StepFunctionExecution
+
+import java.time.{Duration, Instant}
+
+class LambdaTest extends AnyFlatSpec with EitherValues:
+
+  val config: Config = Config("", "", "")
+
+  def notImplemented[T]: IO[Nothing] = IO.raiseError(new Exception("Not implemented"))
+
+  def predictableRandomNumberSelector(selected: Int = 10): (Int, Int) => Int = { (ignoredMin, ignoredMax) => if selected > ignoredMax then ignoredMax else selected }
```

Comment:
1. You don't need the outer braces here
2. since your not using `ignoredMin`, you could make it "_"
3. How come it's called "**ignored**Max"? It won't be ignored if `selected` is greater than it, right?